rust/src/proofs/api.rs

```diff
@@ -100,105 +78,51 @@ pub unsafe extern "C" fn fil_write_without_alignment(
     src_size: u64,
     dst_fd: libc::c_int,
 ) -> *mut fil_WriteWithoutAlignmentResponse {
-    catch_panic_response(|| {
-        init_log();
-
-        info!("write_without_alignment: start");
-
-        let mut response = fil_WriteWithoutAlignmentResponse::default();
-
-        match write_and_preprocess(
+    catch_panic_response2("write_without_alignment", || {
+        let (info, written) = write_and_preprocess(
             registered_proof.into(),
             FileDescriptorRef::new(src_fd),
             FileDescriptorRef::new(dst_fd),
             UnpaddedBytesAmount(src_size),
-        ) {
-            Ok((info, written)) => {
-                response.comm_p = info.commitment;
-                response.status_code = FCPResponseStatus::FCPNoError;
-                response.total_write_unpadded = written.into();
-            }
-            Err(err) => {
-                response.status_code = FCPResponseStatus::FCPUnclassifiedError;
-                response.error_msg = rust_str_to_c_str(format!("{:?}", err));
-            }
-        }
+        )?;
 
-        info!("write_without_alignment: finish");
-
-        raw_ptr(response)
+        Ok(fil_WriteWithoutAlignment {
+            comm_p: info.commitment,
+            total_write_unpadded: written.into(),
+        })
     })
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn fil_fauxrep(
     registered_proof: fil_RegisteredSealProof,
-    cache_dir_path: *const libc::c_char,
-    sealed_sector_path: *const libc::c_char,
+    cache_dir_path: fil_Bytes,
+    sealed_sector_path: fil_Bytes,
```

Comment:
`fil_Bytes` will free the underlying memory. Do we need to some form of "borrowed" form of this type that doesn't drop? Not sure if we own these paths or not.

Reply:
for now I am making them all owned, but probably should add a ref version and optimize at least the paths

Reply:
looks like I can just use `&fil_Bytes`